Add error handling code for some external APIs

ZhouyangJia opened the pull request:
<!--- Provide a general summary of your change in the Title above -->
Hi,

I analyzed the source code and find some APIs missing error handling code, which may cause crashes. This change is aimed to fix the potential bugs.

Best,
Zhouyang
## Description
<!--- Describe your change in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch commented:
Your changes looks very good and make really sence. Thanks a lot!

Only try to change directly to new C++11 style for the `NULL` with `nullptr` and for `bind(testSocket, (struct sockaddr*)&m_addr.saddr, m_addr.size)` with `bind(testSocket, static_cast<struct sockaddr*>(&m_addr.saddr), m_addr.size)`.

Question for the others in team, makes it sence to add a `CLog::Log(...)` message there, to see the source of fault?

AlwinEsch commented on xbmc/network/Socket.cpp:
```diff
@@ -69,7 +69,7 @@ bool CPosixUDPSocket::Bind(bool localOnly, int port, int range)
           {
             m_addr.saddr.saddr6.sin6_port = htons(m_iPort);
-            if (bind(testSocket, (struct sockaddr*)&m_addr.saddr, m_addr.size) >= 0)
+            if (bind(testSocket, static_cast<struct sockaddr*>(&m_addr.saddr), m_addr.size) >= 0)
```
Sorry, has testet for my comment now. Should be `reinterpret_cast` otherwise it brings compile fault.

Razzeee commented:
@AlwinEsch yes, I would encourage adding some logging

ZhouyangJia committed:
Add error handling for fopen

When fopen fails, the program will crash at "while (fgets(buf, sizeof (buf), fp))" since fp is null.

ZhouyangJia committed:
Add error handling for socket

Check the return value of socket to void unexpected result.

AlwinEsch commented:
From my side looks OK to merge and become in (only the NULL still there ;-) )

phunkyfish commented:
@ZhouyangJia do you want to fix the final comment from @AlwinEsch so this can be merged?

AlwinEsch commented:
Yes, the NULL and also a bit logging. To have this fix request whould be good.

ZhouyangJia commented:
Hi, I fixed these issues in #17458 

Sorry for the late reply.
